pygmt/src/grdtrack.py

```diff
@@ -248,7 +247,7 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
         - None if ``outfile`` is set (track output will be stored in file set
           by ``outfile``)
     """
-    if data_kind(points) == "matrix" and newcolname is None:
+    if isinstance(points, pd.DataFrame) and newcolname is None:
```

Comment:
Minor detail, but I'd prefer checking if the `points` variable has a `columns` attribute so that `pandas.DataFrame`, [`cudf.DataFrame`](https://docs.rapids.ai/api/cudf/stable/api.html#dataframe) and [`dask.DataFrame`](https://docs.dask.org/en/stable/dataframe-api.html#dask.dataframe.DataFrame) objects (or any PyData object which implements 'column') will work.

```suggestion
    if hasattr(points, "columns") and newcolname is None:
```

```python
import numpy as np
import pandas as pd
import dask.dataframe as dd
import cudf

numpy_points = np.array([[0.1, 0.2, 0.3]])
hasattr(numpy_points, "columns")  # False
pandas_points = pd.DataFrame(data=[[0.1, 0.2, 0.3]], columns=["x", "y", "z"])
hasattr(pandas_points, "columns")  # True
dask_points = dd.from_pandas(data=pandas_points, npartitions=1)
hasattr(dask_points, "columns")  # True
cudf_points = cudf.DataFrame(data=[[0.1, 0.2, 0.3]], columns=["x", "y", "z"])
hasattr(cudf_points, "columns")  # True
```

Reply:
@weiji14 I'm assuming this block of code is something you want added to `test_grdtrack.py`?

If so, how should the test import `dask` and `cudf`, as they are not listed as dependencies in `environment.yml`?

Reply:
Ideally we would test it yes, but the API of `cudf` and `dask` is supposed to mimic pandas so we don''t really need to either...

Reply:
So should I leave it out of the test then and just test with pandas DataFrame?

Reply:
> So should I leave it out of the test then and just test with pandas DataFrame?

yep